Makefile

```diff
@@ -0,0 +1,12 @@
+PORT ?= # Recommended: 127.0.0.1:3000:3000
+DOCKER_USERNAME ?= # Your Docker username
+APPLICATION ?= book-werm
+IMAGE ?= ${DOCKER_USERNAME}/${APPLICATION}
+
+
+build:
+	docker build -t ${IMAGE} .
+
+run:
+	docker run --name ${APPLICATION} -d -p ${PORT} ${IMAGE}
+	@echo "Running ${IMAGE} on ${PORT}, or open through Docker Desktop"
```

Comment:
Does this work without `PORT` and `DOCKER_USERNAME` filled out?

Reply:
Good point, will add logic to default that command when running without `PORT` or `DOCKER_USERNAME`.

Reply:
Ended up just setting default values that can be changed whenever by the developer for local testing.

Reply:
There are some other options you can do for default values. Maybe use an environment variable that defaults to an empty string or maybe the Makefile itself can set the environment variable to a default value. But this should work for now.